flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunnerConfigurationTest.java

```diff
@@ -0,0 +1,186 @@
+		}
+	}
+
+	@Test
+	public void testTaskManagerRpcServiceShouldBindToAddressDeterminedByConnectingToResourceManager() throws Exception {
+		final ServerSocket testJobManagerSocket = openServerSocket();
+		final Configuration config = createFlinkConfigWithJobManagerPort(testJobManagerSocket.getLocalPort());
+		final HighAvailabilityServices highAvailabilityServices = createHighAvailabilityServices(config);
+
+		RpcService taskManagerRpcService = null;
+		try {
+			taskManagerRpcService = TaskManagerRunner.createRpcService(config, highAvailabilityServices);
+			assertThat(taskManagerRpcService.getAddress(), not(isEmptyOrNullString()));
+		} finally {
+			maybeCloseRpcService(taskManagerRpcService);
+			highAvailabilityServices.closeAndCleanupAllData();
+			IOUtils.closeQuietly(testJobManagerSocket);
+		}
+	}
+
+	@Test
+	public void testCreatingTaskManagerRpcServiceShouldFailIfRpcPortRangeIsInvalid() throws Exception {
+		final Configuration config = new Configuration(createFlinkConfigWithPredefinedTaskManagerHostname("example.org"));
+		config.setString(TaskManagerOptions.RPC_PORT, "-1");
+
+		final HighAvailabilityServices highAvailabilityServices = createHighAvailabilityServices(config);
+
+		try {
+			TaskManagerRunner.createRpcService(config, highAvailabilityServices);
+			fail("Should fail because -1 is not a valid port range");
+		} catch (final IllegalArgumentException e) {
+			assertThat(e.getMessage(),  containsString("Invalid port range definition: -1"));
+		} finally {
+			highAvailabilityServices.closeAndCleanupAllData();
+		}
+	}
+
+	@Test
+	public void testDefaultFsParameterLoading() throws Exception {
+		try {
+			final File tmpDir = temporaryFolder.newFolder();
+			final File confFile = new File(tmpDir, GlobalConfiguration.FLINK_CONF_FILENAME);
+
+			final URI defaultFS = new URI("otherFS", null, "localhost", 1234, null, null, null);
+
+			final PrintWriter pw1 = new PrintWriter(confFile);
+			pw1.println("fs.default-scheme: " + defaultFS);
+			pw1.close();
+
+			String[] args = new String[] {"--configDir", tmpDir.toString()};
+			Configuration configuration = TaskManagerRunner.loadConfiguration(args);
+			FileSystem.initialize(configuration);
+
+			assertEquals(defaultFS, FileSystem.getDefaultFsUri());
+		}
+		finally {
+			// reset FS settings
+			FileSystem.initialize(new Configuration());
+		}
+	}
+
+	private static Configuration createFlinkConfigWithPredefinedTaskManagerHostname(
+			final String taskmanagerHost) {
+		final Configuration config = new Configuration();
+		config.setString(TaskManagerOptions.HOST, taskmanagerHost);
+		config.setString(JobManagerOptions.ADDRESS, "localhost");
+		return new UnmodifiableConfiguration(config);
+	}
+
+	private static Configuration createFlinkConfigWithJobManagerPort(final int port) {
+		Configuration config = new Configuration();
+		config.setString(JobManagerOptions.ADDRESS, "localhost");
+		config.setInteger(JobManagerOptions.PORT, port);
+		return config;
```

Comment:
Maybe this could be symmetric to `createFlinkConfigWithPredefinedTaskManagerHostname` and also return an `UnmodifiableConfiguration`.